Source/Core/UICommon/DiscordPresence.cpp

```diff
@@ -244,6 +249,19 @@ void UpdateDiscordPresence(int party_size, SecretType type, const std::string& s
       // Note: joining still works without partyMax
     }
   }
+#ifdef USE_RETRO_ACHIEVEMENTS
+  else if (Config::Get(Config::RA_ENABLED) && Config::Get(Config::RA_RICH_PRESENCE_ENABLED) &&
+           Config::Get(Config::RA_DISCORD_PRESENCE_ENABLED))
+  {
+    state_string = AchievementManager::GetInstance().GetRichPresence().data();
+    if (state_string.length() >= 128)
```

Comment:
Does it make sense for this to go here?  Is the achievement's "RichPresence" different from dicords?  If this is the same, why does the achievement define a max size of `RP_SIZE` (256) when there can only ever be 124 characters?

Is it just text or are there special pieces of metadata as well?  If so, this simple append '...' will have issues..

Reply:
RetroAchievements has its own 256-char rich presence that is used internally and on the RetroAchievements website. It is just text and this is the solution PCSX2 uses, for example.

Reply:
Ok, thanks, so it probably makes sense to keep this logic in the DiscordPresence.

Also, I would like to know about this:

> Is it just text or are there special pieces of metadata as well?

Reply:
To the best of my knowledge, it is just text. (Said that in the last comment but I guess it was a little subtle :P)